### PR TITLE
Desktop layout for create and edit discussion

### PR DESCRIPTION
**Pull request by joaojusto**
When on desktop, the create and edit discussion form is shown in one page instead.

screen shot 2015-01-15 at 23 44 12
screen shot 2015-01-15 at 23 44 29

### CODE REVIEW

File: app/assets/javascripts/app/discussions/discussion_form.js [hunk 1]
```diff
@@ -12,7 +12,12 @@
     ctrl.discussion = discussion;
     ctrl.saveDiscussion = saveDiscussion;
 
-    $state.go($state.current.name + '.step1');
+    var mobileDetect = new MobileDetect(window.navigator.userAgent);
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/discussion_form.js]: This needs to be hidden inside a service.

File: app/assets/javascripts/app/core/mobile_detect.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(function() {
+  'use strict';
+
+  angular
+  .module('council.core')
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/core/mobile_detect.js]: This must be indented.

File: app/assets/javascripts/app/core/mobile_detect.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  .factory('mobileDetect', mobileDetect);
+
+  function mobileDetect($window) {
+    var mobileDetectAPI = new MobileDetect($window.navigator.userAgent);
+
+    function isMobile() {
+      return !!mobileDetectAPI.mobile();
+    }
+
+    // debugger;
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/core/mobile_detect.js]: You didn't see this?

File: app/assets/javascripts/app/core/mobile_detect.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+(function() {
+  'use strict';
+
+  angular
+  .module('council.core')
+  .factory('mobileDetect', mobileDetect);
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/core/mobile_detect.js]: It's a service, not a factory, factory is for creating instances. And the factory is called `mobileDetect` and the method is `isMobile`, so the a common usage would be `mobileDetect.isMobile()`, witch is acceptable, not good but acceptable. But what about if you wan't to ask if it's tablet or desktop? `mobileDetect.isDesktop()` doesn't look like a good name.

File: app/assets/javascripts/app/discussions/discussion_form.js [hunk 5]
```diff
@@ -5,17 +5,21 @@
...
+  function DiscussionFormCtrl($state, $mdToast, Discussion, discussion, mobileDetect) {
     var ctrl = this;
 
     ctrl.nextScreen = nextScreen;
     ctrl.discussion = discussion;
+    ctrl.mobileDetect = mobileDetect;
     ctrl.saveDiscussion = saveDiscussion;
 
-    $state.go($state.current.name + '.step1');
+    if(ctrl.mobileDetect.isMobile())
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/discussion_form.js]: The `{ }` are required, please follow the guides.

File: app/assets/javascripts/app/core/mobile_detect.js (hunk 2 above)
Comment (joaojusto) [inline app/assets/javascripts/app/core/mobile_detect.js]: Ok

File: app/assets/javascripts/app/core/mobile_detect.js (hunk 3 above)
Comment (joaojusto) [inline app/assets/javascripts/app/core/mobile_detect.js]: No :disappointed:

File: app/assets/javascripts/app/core/mobile_detect.js (hunk 4 above)
Comment (joaojusto) [inline app/assets/javascripts/app/core/mobile_detect.js]: I didn't understand your point. Can you explain better?

File: app/assets/javascripts/app/discussions/discussion_form.js (hunk 5 above)
Comment (joaojusto) [inline app/assets/javascripts/app/discussions/discussion_form.js]: Ok

File: app/assets/javascripts/app/discussions/discussion_form.js [hunk 6]
```diff
@@ -5,17 +5,23 @@
...
     ctrl.nextScreen = nextScreen;
     ctrl.discussion = discussion;
+    ctrl.mobileDetect = mobileDetect;
     ctrl.saveDiscussion = saveDiscussion;
 
-    $state.go($state.current.name + '.step1');
+    if(ctrl.mobileDetect.isMobile()) {
+      $state.go($state.current.name + '.step1');
+    }
+    else {
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/discussion_form.js]: `} else {`

**Comment by joaojusto** [comment]
@gabrielpoca bump

File: app/assets/javascripts/app/core/mobile_detect.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('council.core')
+    .factory('mobileDetect', mobileDetect);
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/core/mobile_detect.js]: Please read the comments with attention, I already said that's this is not a factory, it's a service, please read [this](https://www.google.pt/search?q=factory+programming&oq=factory+programming&aqs=chrome..69i57.5804j0j7&sourceid=chrome&es_sm=91&ie=UTF-8) . Also, I already commented that `mobileDetect` is not a good name.

File: app/assets/javascripts/app/discussions/discussion_form.js [hunk 8]
```diff
@@ -5,17 +5,22 @@
     .module('council.discussions')
     .controller('DiscussionFormCtrl', DiscussionFormCtrl);
 
-  function DiscussionFormCtrl(Discussion, $state, $mdToast, $scope, discussion) {
+  function DiscussionFormCtrl($state, $mdToast, Discussion, discussion, mobileDetect) {
     var ctrl = this;
 
     ctrl.nextScreen = nextScreen;
     ctrl.discussion = discussion;
+    ctrl.mobileDetect = mobileDetect;
```
Comment (gabrielpoca) [inline app/assets/javascripts/app/discussions/discussion_form.js]: Why do you need this?

**Comment by gabrielpoca** [comment]
:+1:

**Pull request merged**